packages/web-components/src/components/va-featured-content/va-featured-content.tsx

```diff
@@ -16,8 +16,19 @@ export class VaFeaturedContent {
    */
   @Prop() uswds?: boolean = false;
 
+  /**
+   * Local state for slot=headline's text.
+   * Used to place an aria-label for role="region" with the same text as the heading.
+   */
+  @State() headlineText: string = null;
+
   @Element() el: HTMLElement;
 
+  componentWillLoad() {
+    let childElements = Array.from(this.el.children);
+    this.headlineText = childElements.find(element => element.slot === "headline").textContent.trim();
+  }
+
```

Comment:
I like this approach but I'm curious about `componentWillLoad`. I was looking at the [available lifecycles](https://stenciljs.com/docs/component-lifecycle) I was thinking maybe not all of the component content would be loaded at this point. I guess it does though? Is `componentDidLoad` is maybe too late for the aria assignment?

Reply:
> I like this approach but I'm curious about `componentWillLoad`. I was looking at the [available lifecycles](https://stenciljs.com/docs/component-lifecycle) and I was thinking maybe not all of the component content would be loaded at this point. I guess it does though? Is `componentDidLoad` maybe too late for the aria assignment?

@jamigibbs Yeah, the `aria-label` was not present in the DOM when I put this in `componentDidLoad`. For `componentWillLoad`, I think that's correct that there's no guarantee that all content will be loaded. My assumption is that because it is static content in the slot, it would be present, but I have nothing to back that up, so I think being on the safer side is better. In the docs it says this can return a promise to wait for data. I added it and just to test it out I added a timeout to ensure it functions as expected. So with this, if the content is already there, great, promise resolves immediately. If for some reason it isn't, it will wait for it before rendering.

```
// setTimeout added just to test that it waits before rendering, removed from final code
  componentWillLoad() {
    return new Promise<void>((resolve) => {
      let childElements = Array.from(this.el.children);
      this.headlineText = childElements.find(element => element.slot === "headline").textContent.trim();
      setTimeout(() => {
        resolve();
      }, 5000);
    });
  }
```
 
Do you think that will work?

Reply:
> Yeah, the aria-label was not present in the DOM when I put this in componentDidLoad.

ah! Interesting. Did using `@State` as you originally had it https://github.com/department-of-veterans-affairs/component-library/pull/1011/commits/773919d71dcc46c4100f328a64cf2488d70202c3  work fine without needing to add this extra `promise` logic? If so it might make more sense to go back to that just to keep things simpler. 🤔

Reply:
> > Yeah, the aria-label was not present in the DOM when I put this in componentDidLoad.
> 
> ah! Interesting. Did using `@State` as you originally had it [773919d](https://github.com/department-of-veterans-affairs/component-library/commit/773919d71dcc46c4100f328a64cf2488d70202c3) work fine without needing to add this extra `promise` logic? If so it might make more sense to go back to that just to keep things simpler. 🤔

Yeah, it worked fine with `@State` or the private var, but I guess the question is whether that DOM content is present on `componentWillLoad`. It seems to be, so I agree, let's go back and keep it simple. If for some reason it stops adding that `aria-label`, we'll know what the problem is.